web/src/routes/datasets/Datasets.tsx

```diff
@@ -71,9 +96,21 @@ class Datasets extends React.Component<DatasetsProps> {
               </Box>
             ) : (
               <>
-                <Box p={2}>
+                <Box paddingTop={2} paddingLeft={2}>
                   <MqText heading>{i18next.t('datasets_route.heading')}</MqText>
                 </Box>
+                <Box className={classes.searchIcon}>
+                  <FontAwesomeIcon icon={faFilter} color={THEME_EXTRA.typography.disabled} />
+                </Box>
+                <Box>
+                  <MqInputBase 
+                    type='text' 
+                    value={this.state.searchTerm}
+                    placeholder='Filter by name' 
+                    id={'filterBar'} 
+                    onChange={string => this.handleSearch(string)}
+                  />
+                </Box>
```

Comment:
Can you put this in a flex box with 'justify-content': 'space-between', 'align-items': 'center' to make it a single line with datasets.